internal/processmon/processmon.go

```diff
@@ -0,0 +1,379 @@
+	// The contextID is optional and is primarily used by remote enforcer
+	// processes to represent the namespace in which the process was running
+	contextID  string
+	exitStatus error
+}
+
+func init() {
+	// Setup new launcher
+	newProcessMon(netNSPath)
+}
+
+// contextID2SocketPath returns the socket path to use for a givent context
+func contextID2SocketPath(contextID string) string {
+
+	if contextID == "" {
+		panic("contextID is empty")
+	}
+	return filepath.Join("/var/run/", contextID+".sock")
+}
+
+// processIOReader will read from a reader and print it on the calling process
+func processIOReader(fd io.Reader, contextID string, exited chan int) {
+
+	reader := bufio.NewReader(fd)
+
+	for {
+		str, err := reader.ReadString('\n')
+		if err != nil {
+			exited <- 1
+			return
+		}
+
+		fmt.Print("[" + contextID + "]:" + str)
+	}
+}
+
+// newProcessMon is a method to create a new processmon
+func newProcessMon(netns string) ProcessManager {
+
+	launcher = &processMon{
+		netNSPath:       netns,
+		activeProcesses: cache.NewCache(processMonitorCacheName),
+		childExitStatus: make(chan exitStatus, 100),
+	}
+
+	go launcher.collectChildExitStatus()
+
+	return launcher
+}
+
+// GetProcessManagerHdl returns a process manager handle.
+func GetProcessManagerHdl() ProcessManager {
+
+	return launcher
+}
+
+// collectChildExitStatus is an async function which collects status for all launched child processes
+func (p *processMon) collectChildExitStatus() {
+
+	for {
+
+		es := <-p.childExitStatus
+		zap.L().Debug("Remote enforcer exited",
+			zap.String("nativeContextID", es.contextID),
+			zap.Int("pid", es.process),
+			zap.Error(es.exitStatus),
+		)
+	}
+}
+
+// SetupLogAndProcessArgs setups args that should be propagated to child processes
+func (p *processMon) SetupLogAndProcessArgs(logToConsole bool, args []string) {
+
+	p.logToConsole = logToConsole
+	p.launcProcessArgs = args
+}
+
+// KillProcess sends a rpc to the process to exit failing which it will kill the process
+func (p *processMon) KillProcess(contextID string) {
+
+	s, err := p.activeProcesses.Get(contextID)
+	if err != nil {
+		zap.L().Debug("Process already killed or never launched")
+		return
+	}
+	req := &rpcwrapper.Request{}
+	resp := &rpcwrapper.Response{}
+	req.Payload = s.(*processInfo).process.Pid
+
+	c := make(chan error, 1)
+	go func() {
+		c <- s.(*processInfo).RPCHdl.RemoteCall(contextID, "Server.EnforcerExit", req, resp)
+	}()
+	select {
+	case kerr := <-c:
+		if kerr != nil {
+			zap.L().Debug("Failed to stop gracefully",
+				zap.String("Remote error", kerr.Error()))
+		}
+
+		if perr := s.(*processInfo).process.Kill(); perr != nil {
+			zap.L().Debug("Process is already dead",
+				zap.String("Kill error", perr.Error()))
+		}
+
+	case <-time.After(5 * time.Second):
+		if perr := s.(*processInfo).process.Kill(); perr != nil {
+			zap.L().Info("Time out while killing process ",
+				zap.Error(perr))
+		}
+	}
+
+	s.(*processInfo).RPCHdl.DestroyRPCClient(contextID)
```

Comment:
you've already case `s` before. Assign it to a variable and reuse it.

Reply:
done